libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -430,15 +431,30 @@ bool AP_Arming::gps_checks(bool report)
             return false;
         }
 
-        // check AHRS and GPS are within 10m of each other
         const Location gps_loc = gps.location();
         Location ahrs_loc;
         if (AP::ahrs().get_position(ahrs_loc)) {
+            // check AHRS and GPS are within 10m of each other horizontally
             const float distance = gps_loc.get_distance(ahrs_loc);
             if (distance > AP_ARMING_AHRS_GPS_ERROR_MAX) {
                 check_failed(ARMING_CHECK_GPS, report, "GPS and AHRS differ by %4.1fm", (double)distance);
                 return false;
             }
+            int32_t gps_alt_cm;
+            if (!gps_loc.get_alt_cm(Location::AltFrame::ABSOLUTE, gps_alt_cm)) {
```

Comment:
Is there any path where this can actually fail? I get that the API wants the checks, but seems a bit silly to spend flash on strings for failure to convert absolute to absolute altitude :)

Reply:
Maybe we can generalize a series of failure strings in `Location` that we could reuse across multiple call sites? (If we wanted to be really fancy, you could have a three arg version that fills it out for you, but most call sites probably don't benefit from getting a string).

Reply:
Well.. this is a format string, so I can encourage the compiler to reuse the format in this case :-)

I'd prefer not to get too fancy with this change.